rs/tests/crypto/crypto_ingress_verification_test.rs

```diff
@@ -0,0 +1,257 @@
+use ic_system_test_driver::util::{
+    UniversalCanister, block_on, expiry_time, random_ed25519_identity, sign_query, sign_update,
+};
+use ic_types::Time;
+use ic_types::crypto::SignedBytesWithoutDomainSeparator;
+use ic_types::messages::{
+    Blob, Delegation, HttpCallContent, HttpCanisterUpdate, HttpQueryContent, HttpRequestEnvelope,
+    HttpUserQuery, SignedDelegation,
+};
+use ic_universal_canister::wasm;
+use rand::{CryptoRng, Rng};
+use reqwest::{StatusCode, Url};
+use slog::{debug, info};
+
+fn main() -> Result<()> {
+    SystemTestGroup::new()
+        .with_setup(setup)
+        .add_test(systest!(query_request_no_delegations))
+        .execute_from_args()?;
+    Ok(())
+}
+
+pub fn setup(env: TestEnv) {
+    InternetComputer::new()
+        .add_subnet(Subnet::fast_single_node(SubnetType::System))
+        .add_subnet(Subnet::fast_single_node(SubnetType::Application))
+        .setup_and_start(&env)
+        .expect("failed to setup IC under test");
+}
+
+#[derive(Clone, Debug)]
+struct TestInformation {
+    url: Url,
+    canister_id: Principal,
+}
+
+pub fn query_request_no_delegations(env: TestEnv) {
+    let logger = env.logger();
+    let node = env.get_first_healthy_node_snapshot();
+    let agent = node.build_default_agent();
+    let rng = &mut reproducible_rng();
+    block_on({
+        async move {
+            let node_url = node.get_public_url();
+            debug!(logger, "Selected replica"; "url" => format!("{}", node_url));
+
+            let canister =
+                UniversalCanister::new_with_retries(&agent, node.effective_canister_id(), &logger)
+                    .await;
+
+            debug!(
+                logger,
+                "Installed Universal Canister";
+                "canister_id" => format!("{:?}", canister.canister_id())
+            );
+
+            let test_info = TestInformation {
+                url: node_url,
+                canister_id: canister.canister_id(),
+            };
+
+            for delegations in 0..=0 {
+                info!(logger, "Testing request with {} delegations", delegations);
+
+                let identity = random_ecdsa_identity(rng);
+                let (delegations, sender) = gen_delegation_chain(rng, delegations, &identity);
+
+                test_accepted_sender_delegation(&test_info, &delegations, &sender, &identity).await;
+            }
+        }
+    });
+}
+
+pub fn random_ecdsa_identity<R: Rng + CryptoRng>(rng: &mut R) -> Secp256k1Identity {
+    Secp256k1Identity::from_private_key(k256::SecretKey::random(rng))
+}
+
+fn sign_delegation(delegation: Delegation, identity: &impl Identity) -> SignedDelegation {
+    let mut msg = b"\x1Aic-request-auth-delegation".to_vec();
+    msg.extend(&delegation.as_signed_bytes_without_domain_separator());
+    let signature = identity.sign_arbitrary(&msg).unwrap();
+
+    SignedDelegation::new(delegation, signature.signature.unwrap())
+}
+
+fn gen_delegation_chain<R: Rng + CryptoRng>(
```

Comment:
There is a `DelegationChainBuilder` and a `SignedDelegationBuilder` in [rs/validator/http_request_test_utils/src/lib.rs](https://github.com/dfinity/ic/blob/master/rs/validator/http_request_test_utils/src/lib.rs), but not sure if this can be used here (I just stumbled upon it but didn't look in detail yet): might be worth to have a look whether this can be used here, or if it is would make sense to extend these test utils.